lms/djangoapps/course_api/blocks/transformers/blocks_api.py

```diff
@@ -59,6 +62,7 @@ def transform(self, usage_info, block_structure):
         """
         Mutates block_structure based on the given usage_info.
         """
+        BlockCompletionTransformer().transform(usage_info, block_structure)
```

Comment:
@tomaszgy  Adding this new transformer to `BlocksAPITransformer` causes it to generate its data every time a course's block structure is created, which results in one additional MySQL query per course when using the course blocks API. That is why a number of tests are failing.

Since we generally won't need to read this data and it doesn't affect block visibility or anything, I think it would be better to only execute this transformer when its field has been explicitly requested, which would involve adding logic [here](https://github.com/edx/edx-platform/blob/9c4869c1d59899a1e4a0ed2bf9cf92c77acc4447/lms/djangoapps/course_api/blocks/api.py#L52-L67) to only add this new transformer when `completion` is in `requested_fields` (or pass `requested_fields` to this class's init method and keep the logic here).

@nasthagiri does that make sense?

Reply:
+1